datalad/support/tests/test_annexrepo.py

```diff
@@ -1134,7 +1134,8 @@ def test_annex_copy_to(origin, clone):
     repo.get("test-annex.dat")
     # now it has:
     eq_(repo.copy_to("test-annex.dat", "target"), ["test-annex.dat"])
-    eq_(repo.copy_to(["INFO.txt", "test-annex.dat"], "target"), ["test-annex.dat"])
+    # and will not be copied again since it was already copied
+    eq_(repo.copy_to(["INFO.txt", "test-annex.dat"], "target"), [])
```

Comment:
1. we should move to use `--json` mode anyways -- I have some unfinished branch already to introduce progress reporting
2. tests still pass -- I can "fix" or refactor envisioning only the cases covered by tests.  So "swallow failures" behavior was never tested I guess?
3. if there is something is not working in `--json` mode we must not just stay with non-json mode, but ask joey to provide desired behavior.  Now there is  http://git-annex.branchable.com/bugs/copy_does_not_reflect_some_failed_copies_in_--json_output/?updated   -- what other errors you experienced besides non-existing files ? note that exit status is non-0 with --json or not, and in our case prior code would fail before even calling out to git-annex:
```shell
$> python -c 'from datalad.api import Dataset; ds = Dataset("."); ds.repo.copy_to(["nonexisting"], "neurosmaug")'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/yoh/proj/datalad/datalad/datalad/support/gitrepo.py", line 296, in newfunc
    result = func(self, files_new, *args, **kwargs)
  File "/home/yoh/proj/datalad/datalad/datalad/support/annexrepo.py", line 1994, in copy_to
    self.get_file_key(files[0])
  File "/home/yoh/proj/datalad/datalad/datalad/support/gitrepo.py", line 296, in newfunc
    result = func(self, files_new, *args, **kwargs)
  File "/home/yoh/proj/datalad/datalad/datalad/support/annexrepo.py", line 953, in get_file_key
    raise IOError(e.code, "File not found.", files)
IOError: [Errno 1] File not found.: 'nonexisting'
```
so whatever errors you expect to be caught -- we better hard-coded expected behavior in the tests!